core/string/translation.cpp

```diff
@@ -1252,9 +1271,23 @@ StringName TranslationServer::tool_translate(const StringName &p_message, const
 	if (tool_translation.is_valid()) {
 		StringName r = tool_translation->get_message(p_message, p_context);
 		if (r) {
+#ifdef TOOLS_ENABLED
+			if (EditorSettings::get_singleton()) {
+				if (EditorSettings::get_singleton()->get("interface/editor/enable_debugging_pseudolocalization")) {
+					return tool_pseudolocalize(r);
+				}
+			}
```

Comment:
This should be cached, it's too costly to query the editor settings again for every single message that will be translated, each time a UI element is refreshed.

I would instead check this in the `setup()` phase like for the normal project, and then do:
```
return editor_pseudolocalization ? tool_pseudolocalize(r) : r;
```